plugins/removeDeprecatedAttrs.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const { attrsGroupsDeprecated, elems } = require('./_collections');
+
+exports.name = 'removeDeprecatedAttrs';
+exports.description = 'removes deprecated attributes';
+
+/**
+ * Remove deprecated attributes.
+ *
+ * @type {import('./plugins-types').Plugin<'removeDeprecatedAttrs'>}
+ */
+exports.fn = (root, params) => {
```

Comment:
To play it extra safe, could you please collect the stylesheet, and only remove the attribute if a selector doesn't traverse it?

For example, let's not remove `clip` if there is a CSS selector that targets `[clip]` or `[clip=…]`. Welcome to introduce a `force` param to bypass this, but I don't think there's any need.

I think the best way to approach this would be:

* Initialize the stylesheet rules with `collectStylesheet(root)`.
* Collect a collection of attribute selectors found in the stylesheet upfront. You can reference [`#includesAttrSelector`](https://github.com/svg/svgo/blob/main/lib/style.js#L297) for how to check for attributes.
* Either skip checking for the attributes in the collection above, or continue checking but on match, also verify it's not in the collection before actually deleting the attribute.

If this is confusing, please let me know and I can elaborate or just handle it for in a commit after other conversations have been resolved.